chains/ethereum/chain.go

```diff
@@ -116,13 +117,19 @@ func InitializeChain(chainCfg *core.ChainConfig, logger log15.Logger, sysErr cha
 	if err != nil {
 		return nil, err
 	}
-	err = conn.EnsureHasBytecode(cfg.erc20HandlerContract)
-	if err != nil {
-		return nil, err
+
+	if cfg.erc20HandlerContract != utils.ZeroAddress {
+		err = conn.EnsureHasBytecode(cfg.erc20HandlerContract)
+		if err != nil {
+			return nil, err
+		}
 	}
-	err = conn.EnsureHasBytecode(cfg.genericHandlerContract)
-	if err != nil {
-		return nil, err
+
+	if cfg.genericHandlerContract != utils.ZeroAddress {
+		err = conn.EnsureHasBytecode(cfg.genericHandlerContract)
+		if err != nil {
+			return nil, err
+		}
 	}
```

Comment:
Both checks seem preferable to the previous if/else chain. I suppose since both of these checks are asking to ensure no `0x` address is found then you could just combine them under one larger conditional:
```go
	if cfg.erc20HandlerContract != utils.ZeroAddress &&
		cfg.genericHandlerContract != utils.ZeroAddress { ... }
```